### DISCUSSION TITLE
Can i create a workflow which runs on all of by self-hosted runners ?

### DISCUSSION THREAD

**Discussion by jeran7x** [community/community#26946] [Actions]
<p>Is there a way to run a workflow say a cleanup or a setup task to run on all of my available self-hosted actions runner. Not just the first one which is available.</p>
<p>If there is no option right now, then can we have that as a feature request please. Think this would be an use-case for most people using self-hosted runners.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/jeran7x">@jeran7x</a> ,</p>
<p>The self-hosted runners will be invoked depending on your job load.</p>
<p>You can create a common/same <a href="https://help.github.com/en/actions/hosting-your-own-runners/using-labels-with-self-hosted-runners#creating-a-custom-label" rel="nofollow noopener">custom label</a> for your self-hosted runners, use matrix to define how many runners you want to invoke.</p>
<p>Note: if you’d like all of runner, specify more(or same) matrix values than runner number.</p>
<p>code sample as below, if you have 3 self-hosted runners, it will invoke them all.</p>
<pre><code>jobs:
  job1:
   runs-on: [label-1] # common label for my self-hosted runners.
   strategy:
     matrix:
       testversion: [3.7, 3.8, 3.9]
   steps:
      - name: test
        run: |
          echo ${{ matrix.testversion }}
</code></pre>
<p>Currently there’s no option to define jobs running all your jobs, if you’d like the function, you can raise a feature_request ticket <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a> where github product manager will take a review.</p>
<p>Thanks.</p>

**Comment by jeran7x** [comment]
<p>I am ok with the workaround we have for now. Thanks this helps my case :slight_smile:</p>
